### PR TITLE
[JP Social/Pre-publish] Integrate real data in the pre-publish home bottom sheet

### PR DESCRIPTION
**Pull request by thomashorta**
Part of #18876 
Needs to be merged only after #18883

## Note to reviewers / testers
You might notice the UI in the Post Settings looks a bit odd, but that's because the UI fixes there are not yet merged. These fixes are available here: https://github.com/wordpress-mobile/WordPress-Android/pull/18884

## To test
Ideally, you need several sites to test the several different scenarios, such as:
- WPCOM site with at least 1 social connection (no share limit)
- Self-hosted with JP Connected and JP Social installed and at least 1 social connection (share limits)
- Site with JP Social but no social connections

Then you can do these steps with each of them:
1. Open app
2. Enable the `FeatureInDevelopment` for Jetpack Social in Debug Settings
3. Select the site
4. Go to `Posts`
5. Create a new post
6. Go to the `Post Settings` and check what the `Social Sharing` section looks like
7. Hit `publish`
8. **Verify** the Social item matches the `Post Settings` state
9. Go back and change some `Social Sharing` options (if possible)
10. Hit `publish`
11. **Verify** the Social item matches the `Post Settings` state again

Try changing the settings to achieve several states (sharing to 0, 1, 2+ networks). Also make sure that hitting `not now` button for the "no connections" UI (when the site has no social connections) only affects the visibility of that flow, e.g.: hiding the item in the "pre-publish" sheet should not hide the section inside the post settings and vice-versa.

## Regression Notes
1. Potential unintended areas of impact
Broken state in Post Settings.

13. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing.

14. What automated tests I added (or what prevented me from doing so)
Updated tests.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] Talkback.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] Large and small screen sizes. (Tablet and smaller phones)
- [ ] Multi-tasking: Split screen and Pop-up view. (Android 10 or higher)

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/PrepublishingHomeViewModel.kt [hunk 1]
```diff
@@ -256,4 +238,17 @@ class PrepublishingHomeViewModel @Inject constructor(
...
         _onActionClicked.postValue(Event(actionType))
     }
+
+    fun updateJetpackSocialState(
+        state: EditorJetpackSocialViewModel.JetpackSocialUiState?
+    ) {
+        val newState = state?.let {
+            SocialUiState.Visible(
+                it,
+                onItemClicked = { Log.d("thomashorta", "hey there!") }
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/PrepublishingHomeViewModel.kt:248]: ❓ Just wanted to confirm that we should merge this because it's going to be addressed in another PR.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/social/PostSocialSharingModelMapper.kt [hunk 2]
```diff
@@ -75,19 +65,20 @@ class PostSocialSharingModelMapper @Inject constructor(
...
-        }.map { (iconResId, connection) ->
+        connections.map { connection ->
+            val iconResId = PublicizeServiceIcon.fromServiceId(connection.service)?.iconResId
             TrainOfIconsModel(
                 data = iconResId,
-                alpha = if (connection.isSharingEnabled) 1f else 0.5f
+                alpha = if (connection.isSharingEnabled) 1f else 0.3f
             )
-        }
+        }.sortedBy { it.alpha }
```
Comment (RenanLukas) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/social/PostSocialSharingModelMapper.kt:74]: 👍

**Review by RenanLukas — APPROVED** [review]
Thanks for the PR, @thomashorta . I've tested with `.com` and SH + JP sites, with and without connections and the states in pre-publishing matched the states in post settings.

File: WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/PrepublishingHomeViewModel.kt (hunk 1 above)
Comment (thomashorta) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/prepublishing/home/PrepublishingHomeViewModel.kt:248]: I think it's fine merging it since this all is behind a Feature Flag anyway, and the PR removing it is already up for review.
